### PR TITLE
Update the VM resource - handle the disks attribute during update

### PR DESCRIPTION
**Pull request by Praveengostu**

### CODE REVIEW

File: ibm/resource_ibm_compute_vm_instance.go [hunk 1]
```diff
@@ -1170,7 +1171,30 @@ func resourceIBMComputeVmInstanceUpdate(d *schema.ResourceData, meta interface{}
 		upgradeOptions[product.NICSpeedCategoryCode] = float64(d.Get("network_speed").(int))
 	}
 
+	if d.HasChange("disks") {
+		oldDisks, newDisks := d.GetChange("disks")
+		oldDisk := oldDisks.([]interface{})
+		newDisk := newDisks.([]interface{})
+		//Remove is not supported
+		if len(oldDisk) > len(newDisk) {
```
[the remark this answers is not in the document]
Comment (ashishth09) [reply ibm/resource_ibm_compute_vm_instance.go]: How about
```
for i, v := range oldDisk {
        if v.(int) != newDisk[i].(int) {
            return fmt.Errorf("Removing drives is not supported. To remove a drive, please create a new support ticket.")
        }
    }
    for i := len(oldDisk); i < len(newDisk); i++ {
        diskName := fmt.Sprintf("guest_disk%d", i)
        capacity := newDisk[i].(int)
        upgradeOptions[diskName] = float64(capacity)
    }
```

File: ibm/structures.go [hunk 2]
```diff
@@ -647,3 +647,13 @@ func flattenGatewayMembers(d *schema.ResourceData, list []datatypes.Network_Gate
 	}
 	return members
 }
+
+func flattenDisks(blkDevices []datatypes.Virtual_Guest_Block_Device) []int {
+	var out = make([]int, 0)
+	for _, v := range blkDevices {
+		if *v.Device != "1" && *v.Device != "7" {
```
Comment (ashishth09) [inline ibm/structures.go]: Can we put comment on why this condition is required?

File: ibm/resource_ibm_compute_vm_instance.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Praveengostu) [reply ibm/resource_ibm_compute_vm_instance.go]: An update to the drive is supported by changing the value also say [10,20,25] - > [20,30,35] but removal of a drive is not supported which is by changing the value also say [10,20,25] - > [10,20]

[the remark this answers is not in the document]
Comment (ashishth09) [reply ibm/resource_ibm_compute_vm_instance.go]: right.. so one optimization is what you have earlier.. which is just comparing  the length

[the remark this answers is not in the document]
Comment (Praveengostu) [reply ibm/resource_ibm_compute_vm_instance.go]: Yes, But any how When rechecking on portal i see there is a way we can detach the disk. But not sure why there is also a not contradicting that removing the drive is via support ticket.. Checking it..

**Comment by Praveengostu** [comment]
UAT result
The ID is 47618217
--- PASS: TestAccIBMComputeVmInstance_basic (742.30s)
=== RUN   TestAccIBMComputeVmInstance_basic_import
--- PASS: TestAccIBMComputeVmInstance_basic_import (252.38s)
PASS
ok  	github.com/terraform-providers/terraform-provider-ibm/ibm	996.424s

**Pull request closed without merging**